bootstrap.sh

```diff
@@ -61,6 +65,55 @@ function self_update_if_its_time() {
     fi
 }
 
+function fail {
+  echo $1 >&2
+  exit 1
+}
+
+function retry {
+  local n=1
+  local max=10
+  local delay=3
+  while true; do
+    "$@" && break || {
+      if [[ $n -lt $max ]]; then
+        ((n++))
+        echo "Command '$@' failed. Attempt $n/$max."
+        sleep $delay;
+      else
+        fail "The command has failed after $n attempts."
+      fi
+    }
+  done
+}
+
+function setup() {
+    current=$(git status -s ${SETUP_FILES_PATHSPEC} | while read mode file; do echo $(stat -f "%m" $file); done|sort -r | awk 'NR==1 {print; exit}')
+
+    echo "Performing setup..."
+
+    for f in ./setup/*.sh; do
+        echo "Running '${f}'."
+        retry bash "$f" -H
+    done
+    echo "Done."
+    echo ""
+
+    echo ${current} > ${SETUP_COMMIT_STAMP_FILE}
+}
+
+function setup_if_needed() {
+    mkdir -p ./setup
```

Comment:
Ah, I have mixed feelings about this.

On the one hand, I understand that by default, we don't need the `setup` in the repository and there is a paragraph in the documentation saying how to set it up. On the other hand, I might feel kinda awkward seeing that after running `up` command, I see a new folder magically generated in my files tree.

Proposals:
1) Add this folder to the repository – it will indicate that such an option is available even without getting through the documentation. Also, we can add a new point to https://github.com/SwingDev/bootstrap-bootstrap/pull/9/files#diff-04c6e90faac2675aa89e2176d2eec7d8R22 and link to the `## Services pre-run setup` to make it easy to find what it is.
2) Check for folder existence and if it doesn't exist, do nothing.